@here/harp-mapview/lib/text/TextElementState.ts

```diff
@@ -225,6 +244,10 @@ export class TextElementState {
         }
 
         this.m_textRenderState = new RenderState();
+        this.m_textLayoutState = new LayoutState();
+        if (this.element.layoutStyle !== undefined) {
+            this.m_textLayoutState.setPlacement(this.element.layoutStyle);
```

Comment:
Pass the layout style in the constructor and set the placement already there.

Reply:
That is what I wanted to do :) - honestly I was dreaming about it ... but this does not comply with other states defined here (m_textRenderState...) and more importantly with your other comments about objects allocations - too many.
Generally I would opt to have this LayoutState as always defined -> this will simplify things A LOT!.
Then the question arises what we should do in TextElementState.replace() call, which moves states from predecessor? 
- copy by reference !!! No we don't want this,
- copy by ref and make undefined in predecessor - No we can not do it, because must be always defined (as said before),
- make a clone() from predecessor (that is what I have - thus clone() copy() methods in LayoutState),
- should we reset state to Theme Based?

I think that is the MOST IMPORTANT design issue in this PR - if we agree on LayoutState always defined (reference always set and never undefined), then the PR will change and simplify in many places.